pandas/core/series.py

```diff
@@ -203,13 +209,15 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                                          '`data` argument and a different '
                                          '`index` argument.  `copy` must '
                                          'be False.')
-            elif isinstance(data, Categorical):
+
+            elif is_extension_array_dtype(data) and dtype is not None:
                 # GH12574: Allow dtype=category only, otherwise error
-                if ((dtype is not None) and
-                        not is_categorical_dtype(dtype)):
-                    raise ValueError("cannot specify a dtype with a "
-                                     "Categorical unless "
-                                     "dtype='category'")
+                if not data.dtype.is_dtype(dtype):
+                    raise ValueError("Cannot specify a dtype '{}' with an "
```

Comment:
Side note (maybe I should open an issue, if there is none yet), but `is_dtype` will always return True for Categoricals, whatever the categories/ordered attributes, I think? Is this actually what we want? As currently doing `pd.Series(pd.Categorical(['a', 'b']), dtype=CategoricalDType(categories=['A', 'B']))` will just ignore the `dtype` instead of raising an error / coercing.

Reply:
I thought there was an issue about this recently, though I couldn't find it easily.

This could be a hole in our ExtensionDtype API. We have `is_dtype` for seeing if the dtypes are the same kind, but we might want an `is_dtype_equal` for checking parametrized dtypes.

https://github.com/pandas-dev/pandas/issues/19741

Reply:
I think we discussed/fixed this for `Series[cat].astype(cat)`, see https://github.com/pandas-dev/pandas/pull/18710 and referenced issues. But probably `Series(cat, dtype=cat)` should do something similar? Or should that be more strict.

(I like your proposal to add `is_dtype_equal` to be more strict)

Reply:
this is much nicer

Reply:
What is much nicer?